packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts

```diff
@@ -89,10 +89,18 @@ export default function useFieldValues<FieldType extends TypedField>({
   const handleFormSubmit = () => {
     const submitValue = isRadioFieldComponent
       ? values
-      : fieldsWithHandlers.reduce(
-          (acc, { name, value }) => ({ ...acc, [name]: value }),
-          {}
-        );
+      : fieldsWithHandlers.reduce((acc, { name, value, type }) => {
+          /*
+                For phone numbers pass the first 3 charactes from value as dialCode until we support a dialCode picker
+            */
+          return type === 'phone'
+            ? {
+                ...acc,
+                country_code: value?.substring(0, 3),
+                [name]: value?.substring(3, value.length),
+              }
+            : { ...acc, [name]: value };
+        }, {});
```

Comment:
Would consider creating a meaningfully named utility for the callback, and explicitly using the `phone_number` key in place of `[name]` for the `phone` handling to be clear about what the keys are

Reply:
Synced offline